hail/src/main/scala/is/hail/lir/SimplifyControl.scala

```diff
@@ -107,18 +107,16 @@ class SimplifyControl(m: Method) {
     for (b <- blocks) {
       if (b.first != null &&
         b.first.isInstanceOf[GotoX]) {
-        u.sameSet(
+        u.union(
           blocks.index(b),
           blocks.index(b.first.asInstanceOf[GotoX].L))
       }
     }
 
     val rootFinalTarget = mutable.Map[Int, Block]()
     blocks.indices.foreach { i =>
-      val r = u.find(i)
-      if (r == i) {
```

Comment:
For context without needing to go look at `UnionFind`: `sameSet(Int, Int): Boolean` tests if two elements are in the same partition, while `union(Int, Int): Unit` actually combines the two partitions. So this first loop was doing nothing, and the `(r == i)` test at the bottom was always true.